packages/playground/remote/src/lib/worker-caching.ts

```diff
@@ -0,0 +1,92 @@
+import { isURLScoped } from '@php-wasm/scopes';
+
+export class WorkerCache {
+	readonly cacheNamePrefix = 'playground-cache';
+
+	private cacheName: string;
+
+	constructor(cacheVersion: string) {
+		this.cacheName = `${this.cacheNamePrefix}-${cacheVersion}`;
+	}
+
+	addCache = async (key: Request, response: Response) => {
+		const clonedResponse = response.clone();
+		const cache = await caches.open(this.cacheName);
+		await cache.put(key, clonedResponse);
+	};
+
+	getCache = async (key: Request) => {
+		const cache = caches.open(this.cacheName);
+		return await cache.then((c) => c.match(key, { ignoreSearch: true }));
+	};
+
+	isValidHostname = (url: URL) => {
```

Comment:
There are a lot of valid hostnames for which this will return false. It might be clearer to name the method `shouldCacheForHostname()` or something similar.

Reply:
I renamed it to `shouldCacheUrl`.